frame/contracts/src/exec.rs

```diff
@@ -903,11 +905,19 @@ where
 			// Every non delegate call or instantiate also optionally transfers the balance.
 			self.initial_transfer()?;
 
+			#[cfg(feature = "unsafe-debug")]
+			let code_hash = executable.code_hash().clone();
+			#[cfg(feature = "unsafe-debug")]
+			T::Debug::before_call(&code_hash, entry_point, &input_data);
```

Comment:
Only clone when compiled with debugging. We can also think about not cloning the input at all and leave that to the debugger to store the input if needed. We already pass it in `before_call` and it doesn't change in between.

```suggestion
			#[cfg(feature = "unsafe-debug")]
			let (code_hash, input_clone) = {
			    let code_hash = executable.code_hash().clone();
			    T::Debug::before_call(&code_hash, entry_point, &input_data);
			    (code_hash, input_data.clone())
			};
```

Reply:
I applied the code suggestion. I wouldn't rely on the debugger to store input, since it might make sense to actually implement only `after_call` or have stateless debugger